src/workerd/api/sockets.c++

```diff
@@ -172,22 +171,26 @@ jsg::Ref<Socket> connectImplNoOutputLock(
     jsg::Lock& js, kj::Maybe<jsg::Ref<Fetcher>> fetcher, AnySocketAddress address,
     jsg::Optional<SocketOptions> options) {
 
+  auto& ioContext = IoContext::current();
+  JSG_REQUIRE(!ioContext.isFiddle(), TypeError, "Socket API not supported in web preview mode.");
+
   // Extract the domain/ip we are connecting to from the address.
   kj::String domain;
   bool isDefaultFetchPort = false;
+
   KJ_SWITCH_ONEOF(address) {
     KJ_CASE_ONEOF(str, kj::String) {
       // We need just the hostname part of the address, i.e. we want to strip out the port.
       // We do this using the standard URL parser since it will handle IPv6 for us as well.
-      auto record = JSG_REQUIRE_NONNULL(url::URL::parse(jsg::usv(kj::str("https://", str))),
+      auto input = kj::str("fake://", str);
+      auto record = JSG_REQUIRE_NONNULL(jsg::Url::tryParse(input.asPtr()),
```

Comment:
The `UrlRecord` type is dropped in this PR. Suggest renaming `s/record/url/` for clarity or using an explicit type instead of `auto`.